packages/renderer/src/App.svelte

```diff
@@ -388,7 +388,9 @@ tablePersistence.storage = new PodmanDesktopStoragePersist();
           <TroubleshootingPage />
         </Route>
         <Route path="/extensions" breadcrumb="Extensions" navigationHint="root">
-          <ExtensionList />
+          <ExtensionList searchTerm={decodeURIComponent(meta.query.searchTerm ?? '')} screen={
+            ['installed', 'catalog', 'development'].includes(meta.query.screen) ? meta.query.screen as 'installed' | 'catalog' | 'development' : 'installed'
+          } />
```

Comment:
```suggestion
          <ExtensionList searchTerm={decodeURIComponent(meta.query.searchTerm ?? '')} screen={meta.query.screen} />
```

let's trust the input

Reply:
This does not work, we need to at least use `as`, or the string type won't match the enum one

Reply:
@axel7083 it seems to me that casting without checking here is counterproductive with using an enum type (https://github.com/podman-desktop/podman-desktop/pull/14656#discussion_r2480943410). If we bypass the check here, the enum type defined in `ExtensionList` component is never enforced